WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SiteCreationCreatingFragment.java

```diff
@@ -0,0 +1,283 @@
+        super.onViewCreated(view, savedInstanceState);
+
+        showHomeButton(!mInModalMode);
+
+        if (savedInstanceState == null) {
+            AnalyticsTracker.track(AnalyticsTracker.Stat.SITE_CREATION_CREATING_VIEWED);
+        }
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+
+        if (!mCreationSucceeded) {
+            // connect to the Service. We'll receive updates via EventBus.
+            mServiceEventConnection = new ServiceEventConnection(getContext(), SiteCreationService.class, this);
+        } else {
+            disableUntil(R.id.site_creation_creating_preparing_frontend);
+        }
+    }
+
+    @Override
+    public void onPause() {
+        super.onPause();
+
+        if (mServiceEventConnection != null) {
+            // disconnect from the Service
+            mServiceEventConnection.disconnect(getContext(), this);
+        }
+    }
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+
+        outState.putBoolean(KEY_IN_MODAL_MODE, mInModalMode);
+        outState.putBoolean(KEY_CREATION_FINISHED, mCreationSucceeded);
+        outState.putBoolean(KEY_WEBVIEW_LOADED_IN_TIME, mWebViewLoadedInTime);
+    }
+
+    private void mutateToCompleted(boolean showWebView) {
+        if (isAdded()) {
+            hideActionbar();
+            mProgressContainer.setVisibility(View.GONE);
+            mCompletedContainer.setVisibility(View.VISIBLE);
+            mWebView.setVisibility(showWebView ? View.VISIBLE : View.INVISIBLE);
+            mTadaContainer.setVisibility(showWebView ? View.INVISIBLE : View.VISIBLE);
+        }
+    }
+
+    private PreviewWebViewClient loadWebview() {
+        String siteAddress = "https://" + getArguments().getString(ARG_SITE_SLUG) + ".wordpress.com";
+        PreviewWebViewClient client = new PreviewWebViewClient(siteAddress);
+        mWebView.setWebViewClient(client);
+        mWebView.loadUrl(siteAddress);
+        return client;
+    }
+
+    private static class PreviewWebViewClient extends URLFilteredWebViewClient {
+        private boolean mIsPageFinished;
+
+        PreviewWebViewClient(String siteAddress) {
+            super(siteAddress);
+        }
+
+        @Override
+        public void onPageFinished(WebView view, String url) {
+            super.onPageFinished(view, url);
+
+            mIsPageFinished = true;
+        }
+    }
+
+    private void disableUntil(@IdRes int textViewId) {
+        boolean enabled = false;
+
+        // traverse the array (elements are in "reverse" order already) and disable them until the provided on is reach.
+        //  From that point on, enable the labels found
+        for(TextView tv : mLabels) {
+            if (tv.getId() == textViewId) {
+                enabled = true;
+            }
+
+            tv.setEnabled(enabled);
+        }
+    }
+
+    private void setModalMode(boolean inModalMode) {
```

Comment:
It looks like `inModalMode` is always `false`.   Will it ever be `true`?

Reply:
`mInModalMode` (the variable) is set to true on first appearance of the fragment. Then, `setModalMode()` is used to set it to false only, yes.